src/RestClientVS/Margin/ResponseControl.xaml.cs

```diff
@@ -14,7 +17,43 @@ public async Task SetResponseTextAsync(RequestResult result)
         {
             if (result.Response != null)
             {
-                Control.Text = await result.Response.ToRawStringAsync();
+                var text = new StringBuilder();
+                switch (result.Response.Content.Headers.ContentType.MediaType)
+                {
+                    case "application/json":
+                        var jsonString = await result.Response.Content.ReadAsStringAsync();
+                        text.AppendLine(result.Response.Headers.ToString());
+                        try
+                        {
+                            var jsonObject = JObject.Parse(jsonString);
+                            text.AppendLine(jsonObject.ToString());
+                        }
+                        catch (Exception ex)
+                        {
+                            text.AppendFormat("** {0} : Error parsing JSON ( {1} ), raw content follows. **\r\r", nameof(RestClientVS), ex.GetBaseException().Message);
+                            text.AppendLine(jsonString);
+                        }
+                        Control.Text = text.ToString();
+                        break;
+                    case "application/xml":
```

Comment:
Same as the "json" comment above but with "xml"

Reply:
same reply as above